### PR TITLE
Test: Fix issues on `kubectl.CiliumReport`

### PR DESCRIPTION
**Pull request by eloycoto**
- Fix issues on duplicated data in CiliumReport
- Fix issues with Variadic arguments
- Delete some os.Stdout messages on CiliumReport (Os.Stdout is not in
order with ginkgoWriter)

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>

### CODE REVIEW

**Comment by eloycoto** [comment]
test-me-please

File: test/helpers/kubectl.go [hunk 1]
```diff
@@ -940,36 +940,30 @@ func (kub *Kubectl) CiliumPolicyAction(namespace, filepath string, action Resour
...
 		}
 	}
+
 	kub.DumpCiliumCommandOutput(namespace)
 	kub.GatherLogs()
 	kub.CheckLogsForDeadlock()
-	fmt.Fprint(wr, "===================== EXITING REPORT GENERATION =====================")
-	fmt.Println("===================== EXITING REPORT GENERATION =====================")
-
+	fmt.Fprintln(wr, "===================== EXITING REPORT GENERATION =====================")
```
Comment (ianvernon) [inline test/helpers/kubectl.go]: Please revert removal of `fmt.Println("===================== EXITING REPORT GENERATION =====================")` . It makes it easy to see where tests fail on Jenkins, as it is not immediately obvious.

**Review by ianvernon — CHANGES_REQUESTED** [review]

File: test/helpers/kubectl.go (hunk 1 above)
Comment (eloycoto) [reply test/helpers/kubectl.go]: Umm, the print is in there to the GinkgoWriter. If you don't use the GinkgoWriter the os.Stdout is not in the same order, and some functions are not in the correct place. 

All GinkgoWriter is dump to stdout in the correct order.

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
@ianvernon  I changed the `======== TEST FAILED =================` to use ginko.GinkgoWriter instead of the log.wr and now it works ok, only one is needed

```

breo.acalustra.com ➜  test git:(Debug) ✗ ginkgo --focus "eloy" -v -- --cilium.provision=false
level=info msg="environment variable \"K8S_VERSION\" was not set; setting to default value \"1.10\""
Running Suite: runtime
======================
Random Seed: 1525248234
Will run 1 of 282 specs

SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS
------------------------------
K8sValidatedServicesTest
  eloy
  /home/eloy/.go/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:312
time="2018-05-02T10:04:00+02:00" level=info msg=Starting testName=K8sServiceTest
time="2018-05-02T10:04:00+02:00" level=info msg="Vagrant: running command 'vagrant ssh-config k8s1-1.10'"
time="2018-05-02T10:04:03+02:00" level=debug msg="generated SSHConfig for node k8s1-1.10"
time="2018-05-02T10:04:03+02:00" level=debug msg="done importing ssh config"
time="2018-05-02T10:04:03+02:00" level=debug msg="running command: ls /tmp/"
time="2018-05-02T10:04:04+02:00" level=debug msg="running command: cat ${HOME}/.kube/config"
time="2018-05-02T10:04:04+02:00" level=debug msg="applying /home/vagrant/go/src/github.com/cilium/cilium/test/k8sT/manifests/cilium_ds.yaml" testName=K8sServiceTest
time="2018-05-02T10:04:04+02:00" level=debug msg="running command: kubectl apply -f  /home/vagrant/go/src/github.com/cilium/cilium/test/k8sT/manifests/cilium_ds.yaml"
time="2018-05-02T10:04:10+02:00" level=debug msg="running command: kubectl -n kube-system get pods -l k8s-app=cilium -o json"
        time="2018-05-02T10:04:20+02:00" level=debug msg="running command: kubectl -n kube-system get pods -l k8s-app=kube-dns -o json"
time="2018-05-02T10:04:20+02:00" level=debug msg="applying https://raw.githubusercontent.com/cilium/microscope/master/docs/microscope.yaml" testName=K8sServiceTest
time="2018-05-02T10:04:20+02:00" level=debug msg="running command: kubectl apply -f  https://raw.githubusercontent.com/cilium/microscope/master/docs/microscope.yaml"
time="2018-05-02T10:04:26+02:00" level=debug msg="running command: kubectl -n kube-system get pods -l k8s-app=microscope -o json"
time="2018-05-02T10:04:27+02:00" level=debug msg="running command: kubectl -n kube-system logs --timestamps=true -l k8s-app=cilium --since=26.365771542s --previous"
===================== TEST FAILED =====================
time="2018-05-02T10:04:28+02:00" level=debug msg="running command: kubectl get pods -o wide --all-namespaces"
time="2018-05-02T10:04:28+02:00" level=debug msg="running command: kubectl exec -n kube-system cilium-snls6 -- cilium service list"
time="2018-05-02T10:04:29+02:00" level=debug msg="running command: kubectl exec -n kube-system cilium-snls6 -- cilium policy get"
time="2018-05-02T10:04:30+02:00" level=debug msg="running command: kubectl exec -n kube-system cilium-snls6 -- cilium endpoint list"
time="2018-05-02T10:04:32+02:00" level=debug msg="running command: kubectl exec -n kube-system cilium-v58pq -- cilium service list"
time="2018-05-02T10:04:33+02:00" level=debug msg="running command: kubectl exec -n kube-system cilium-v58pq -- cilium policy get"
time="2018-05-02T10:04:33+02:00" level=debug msg="running command: kubectl exec -n kube-system cilium-v58pq -- cilium endpoint list"
time="2018-05-02T10:04:35+02:00" level=debug msg="running command: kubectl exec -n kube-system cilium-snls6 -- cilium bpf tunnel list"
time="2018-05-02T10:04:36+02:00" level=debug msg="running command: kubectl exec -n kube-system cilium-snls6 -- cilium policy get"
time="2018-05-02T10:04:37+02:00" level=debug msg="running command: kubectl exec -n kube-system cilium-snls6 -- cilium kvstore get cilium --recursive"
time="2018-05-02T10:04:38+02:00" level=debug msg="running command: kubectl exec -n kube-system cilium-snls6 -- cilium endpoint list -o json"
time="2018-05-02T10:04:39+02:00" level=debug msg="running command: kubectl exec -n kube-system cilium-snls6 -- cilium service list -o json"
time="2018-05-02T10:04:40+02:00" level=debug msg="running command: kubectl exec -n kube-system cilium-snls6 -- cilium status --all-controllers"
time="2018-05-02T10:04:42+02:00" level=debug msg="running command: kubectl exec -n kube-system cilium-snls6 -- cilium config"
time="2018-05-02T10:04:44+02:00" level=debug msg="running command: kubectl exec -n kube-system cilium-snls6 -- cilium bpf lb list"
time="2018-05-02T10:04:46+02:00" level=debug msg="running command: kubectl exec -n kube-system cilium-snls6 -- cilium bpf ct list global"
time="2018-05-02T10:04:47+02:00" level=debug msg="running command: kubectl exec -n kube-system cilium-snls6 -- cilium-bugtool"
time="2018-05-02T10:05:17+02:00" level=debug msg="running command: kubectl exec -n kube-system cilium-snls6 -- ls /tmp/"
time="2018-05-02T10:05:18+02:00" level=debug msg="running command: kubectl cp kube-system/cilium-snls6:/tmp/cilium-bugtool-20180502-062547.331+0000-UTC-308092167.tar /tmp/cilium-bugtool-20180502-062547.331+0000-UTC-308092167.tar"
time="2018-05-02T10:05:18+02:00" level=debug msg="running command: cp /tmp/cilium-bugtool-20180502-062547.331+0000-UTC-308092167.tar /home/vagrant/go/src/github.com/cilium/cilium/test/test_results/110-/K8sValidatedServicesTesteloy/cilium-snls6-cilium-bugtool-20180502-062547.331+0000-UTC-308092167.tar"
time="2018-05-02T10:05:19+02:00" level=debug msg="running command: kubectl cp kube-system/cilium-snls6:/tmp/cilium-bugtool-20180502-062754.56+0000-UTC-320490566.tar /tmp/cilium-bugtool-20180502-062754.56+0000-UTC-320490566.tar"
time="2018-05-02T10:05:19+02:00" level=debug msg="running command: cp /tmp/cilium-bugtool-20180502-062754.56+0000-UTC-320490566.tar /home/vagrant/go/src/github.com/cilium/cilium/test/test_results/110-/K8sValidatedServicesTesteloy/cilium-snls6-cilium-bugtool-20180502-062754.56+0000-UTC-320490566.tar"
time="2018-05-02T10:05:19+02:00" level=debug msg="running command: kubectl cp kube-system/cilium-snls6:/tmp/cilium-bugtool-20180502-063052.671+0000-UTC-033770139.tar /tmp/cilium-bugtool-20180502-063052.671+0000-UTC-033770139.tar"
time="2018-05-02T10:05:20+02:00" level=debug msg="running command: cp /tmp/cilium-bugtool-20180502-063052.671+0000-UTC-033770139.tar /home/vagrant/go/src/github.com/cilium/cilium/test/test_results/110-/K8sValidatedServicesTesteloy/cilium-snls6-cilium-bugtool-20180502-063052.671+0000-UTC-033770139.tar"
time="2018-05-02T10:05:20+02:00" level=debug msg="running command: kubectl cp kube-system/cilium-snls6:/tmp/cilium-bugtool-20180502-080448.32+0000-UTC-568267439.tar /tmp/cilium-bugtool-20180502-080448.32+0000-UTC-568267439.tar"
time="2018-05-02T10:05:21+02:00" level=debug msg="running command: cp /tmp/cilium-bugtool-20180502-080448.32+0000-UTC-568267439.tar /home/vagrant/go/src/github.com/cilium/cilium/test/test_results/110-/K8sValidatedServicesTesteloy/cilium-snls6-cilium-bugtool-20180502-080448.32+0000-UTC-568267439.tar"
time="2018-05-02T10:05:21+02:00" level=debug msg="running command: kubectl exec -n kube-system cilium-snls6 -- ls /tmp/ | grep core"
time="2018-05-02T10:05:22+02:00" level=debug msg="There is no core dumps in the pod" pod=cilium-snls6 testName=K8sServiceTest
time="2018-05-02T10:05:22+02:00" level=debug msg="running command: kubectl exec -n kube-system cilium-v58pq -- cilium policy get"
time="2018-05-02T10:05:22+02:00" level=debug msg="running command: kubectl exec -n kube-system cilium-v58pq -- cilium service list -o json"
time="2018-05-02T10:05:23+02:00" level=debug msg="running command: kubectl exec -n kube-system cilium-v58pq -- cilium bpf lb list"
time="2018-05-02T10:05:24+02:00" level=debug msg="running command: kubectl exec -n kube-system cilium-v58pq -- cilium kvstore get cilium --recursive"
time="2018-05-02T10:05:25+02:00" level=debug msg="running command: kubectl exec -n kube-system cilium-v58pq -- cilium endpoint list -o json"
time="2018-05-02T10:05:26+02:00" level=debug msg="running command: kubectl exec -n kube-system cilium-v58pq -- cilium bpf tunnel list"
time="2018-05-02T10:05:26+02:00" level=debug msg="running command: kubectl exec -n kube-system cilium-v58pq -- cilium config"
time="2018-05-02T10:05:27+02:00" level=debug msg="running command: kubectl exec -n kube-system cilium-v58pq -- cilium bpf ct list global"
time="2018-05-02T10:05:28+02:00" level=debug msg="running command: kubectl exec -n kube-system cilium-v58pq -- cilium status --all-controllers"
time="2018-05-02T10:05:29+02:00" level=debug msg="running command: kubectl exec -n kube-system cilium-v58pq -- cilium-bugtool"
time="2018-05-02T10:05:37+02:00" level=debug msg="running command: kubectl exec -n kube-system cilium-v58pq -- ls /tmp/"
time="2018-05-02T10:05:38+02:00" level=debug msg="running command: kubectl cp kube-system/cilium-v58pq:/tmp/cilium-bugtool-20180502-062617.445+0000-UTC-887631117.tar /tmp/cilium-bugtool-20180502-062617.445+0000-UTC-887631117.tar"
time="2018-05-02T10:05:38+02:00" level=debug msg="running command: cp /tmp/cilium-bugtool-20180502-062617.445+0000-UTC-887631117.tar /home/vagrant/go/src/github.com/cilium/cilium/test/test_results/110-/K8sValidatedServicesTesteloy/cilium-v58pq-cilium-bugtool-20180502-062617.445+0000-UTC-887631117.tar"
time="2018-05-02T10:05:38+02:00" level=debug msg="running command: kubectl cp kube-system/cilium-v58pq:/tmp/cilium-bugtool-20180502-063130.406+0000-UTC-899465621.tar /tmp/cilium-bugtool-20180502-063130.406+0000-UTC-899465621.tar"
time="2018-05-02T10:05:39+02:00" level=debug msg="running command: cp /tmp/cilium-bugtool-20180502-063130.406+0000-UTC-899465621.tar /home/vagrant/go/src/github.com/cilium/cilium/test/test_results/110-/K8sValidatedServicesTesteloy/cilium-v58pq-cilium-bugtool-20180502-063130.406+0000-UTC-899465621.tar"
time="2018-05-02T10:05:39+02:00" level=debug msg="running command: kubectl cp kube-system/cilium-v58pq:/tmp/cilium-bugtool-20180502-080529.744+0000-UTC-215711164.tar /tmp/cilium-bugtool-20180502-080529.744+0000-UTC-215711164.tar"
time="2018-05-02T10:05:40+02:00" level=debug msg="running command: cp /tmp/cilium-bugtool-20180502-080529.744+0000-UTC-215711164.tar /home/vagrant/go/src/github.com/cilium/cilium/test/test_results/110-/K8sValidatedServicesTesteloy/cilium-v58pq-cilium-bugtool-20180502-080529.744+0000-UTC-215711164.tar"
time="2018-05-02T10:05:40+02:00" level=debug msg="running command: kubectl exec -n kube-system cilium-v58pq -- ls /tmp/ | grep core"
time="2018-05-02T10:05:40+02:00" level=debug msg="There is no core dumps in the pod" pod=cilium-v58pq testName=K8sServiceTest
time="2018-05-02T10:05:40+02:00" level=debug msg="running command: kubectl get pods --all-namespaces -o json"
time="2018-05-02T10:05:41+02:00" level=debug msg="running command: kubectl -n kube-system logs --timestamps cilium-snls6"
time="2018-05-02T10:05:41+02:00" level=debug msg="running command: kubectl -n kube-system logs --timestamps --previous cilium-snls6"
time="2018-05-02T10:05:41+02:00" level=debug msg="running command: kubectl -n kube-system logs --timestamps --previous cilium-v58pq"
time="2018-05-02T10:05:42+02:00" level=debug msg="running command: kubectl get ds --all-namespaces -o json"
time="2018-05-02T10:05:42+02:00" level=debug msg="running command: kubectl get deployment --all-namespaces -o json"
time="2018-05-02T10:05:42+02:00" level=debug msg="running command: kubectl -n kube-system logs -l k8s-app='kube-dns' -c kubedns"
time="2018-05-02T10:05:43+02:00" level=debug msg="running command: kubectl get services --all-namespaces -o json"
time="2018-05-02T10:05:44+02:00" level=debug msg="running command: kubectl get netpol --all-namespaces -o json"
time="2018-05-02T10:05:44+02:00" level=debug msg="running command: kubectl -n kube-system logs -l k8s-app='kube-dns' -c dnsmasq"
time="2018-05-02T10:05:44+02:00" level=debug msg="running command: kubectl -n kube-system logs --timestamps cilium-v58pq"
time="2018-05-02T10:05:45+02:00" level=debug msg="running command: kubectl get cnp --all-namespaces -o json"
time="2018-05-02T10:05:45+02:00" level=debug msg="running command: kubectl describe pods --all-namespaces"
time="2018-05-02T10:05:45+02:00" level=debug msg="running command: kubectl get replicationcontroller --all-namespaces -o json"
time="2018-05-02T10:05:46+02:00" level=debug msg="running command: kubectl -n kube-system logs -l k8s-app='kube-dns' -c sidecar"
time="2018-05-02T10:05:46+02:00" level=debug msg="running command: kubectl -n kube-system logs --timestamps=true -l k8s-app=cilium | grep -qi -B 5 -A 5 deadlock"
time="2018-05-02T10:05:46+02:00" level=debug msg="running command: kubectl -n kube-system logs --timestamps=true --previous -l k8s-app=cilium | grep -qi -B 5 -A 5 deadlock"
===================== EXITING REPORT GENERATION =====================
time="2018-05-02T10:05:52+02:00" level=debug msg="running command: kubectl get pods --all-namespaces -o jsonpath='{.items[*].metadata.deletionTimestamp}'"
time="2018-05-02T10:05:52+02:00" level=info msg="build is not running on Jenkins; environment variable 'JENKINS_HOME' is not set"
time="2018-05-02T10:05:52+02:00" level=info msg="build is not running on Jenkins; environment variable 'NODE_NAME' is not set"
time="2018-05-02T10:05:52+02:00" level=info msg="AfterSuite: not running on Jenkins; leaving VMs running for debugging"

• Failure [111.621 seconds]
K8sValidatedServicesTest
/home/eloy/.go/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:280
  eloy [It]
  /home/eloy/.go/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:312

  Expected
      <bool>: true
  to be false

  /home/eloy/.go/src/github.com/cilium/cilium/test/k8sT/Services.go:115
------------------------------
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS

Summarizing 1 Failure:

[Fail] K8sValidatedServicesTest [It] eloy
/home/eloy/.go/src/github.com/cilium/cilium/test/k8sT/Services.go:115

Ran 1 of 282 Specs in 111.625 seconds
FAIL! -- 0 Passed | 1 Failed | 0 Pending | 281 Skipped --- FAIL: TestTest (111.64s)
FAIL
```

The test that fails is a sample test: 
```
diff --git a/test/k8sT/Services.go b/test/k8sT/Services.go
index 0f0e9f08f..0e7e06c0b 100644
--- a/test/k8sT/Services.go
+++ b/test/k8sT/Services.go
@@ -111,7 +111,9 @@ var _ = Describe("K8sValidatedServicesTest", func() {
                        ExpectWithOffset(1, err).Should(BeNil())
                }
        }
-
+       It("eloy", func() {
+               Expect(true).To(BeFalse())
+       })
        Context("Checks ClusterIP Connectivity", func() {
 
                var (

```

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
Hit the bookinfo issue, commit was not rebased with master: 

https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/2885/testReport/junit/k8s-1/7/K8sValidatedServicesTest_Bookinfo_Demo_Tests_bookinfo_demo/

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Review by ianvernon — APPROVED** [review]

**Pull request merged**